Dockerfile

```diff
@@ -1,50 +1,60 @@
 # This Dockerfile can be used to build a Docker image suitable for tango projects.
 
-# It's important to use the right base image depending on the environment you're going
-# to run your project in. If you need different version of CUDA, you'll have to change the base
-# image accordingly and also potentially adjust the PyTorch install line below.
-# See https://hub.docker.com/r/nvidia/cuda/tags for a list of available base images.
-# Generally the "runtime" images are best suited for tango projects.
-ARG cuda=11.5.1
-FROM nvidia/cuda:${cuda}-cudnn8-runtime-ubuntu20.04
+ARG BASE_IMAGE=ubuntu:18.04
+ARG PYTHON_VERSION=3.9
+
+FROM ${BASE_IMAGE} as dev-base
+RUN --mount=type=cache,id=apt-dev,target=/var/cache/apt \
+    apt-get update && apt-get install -y --no-install-recommends \
+        build-essential \
+        ca-certificates \
+        curl \
+        git && \
+    rm -rf /var/lib/apt/lists/*
+ENV PATH /opt/conda/bin:$PATH
+
+FROM dev-base as conda
+ARG PYTHON_VERSION=3.9
+RUN curl -fsSL -v -o ~/miniconda.sh -O  https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh  && \
+    chmod +x ~/miniconda.sh && \
+    ~/miniconda.sh -b -p /opt/conda && \
+    rm ~/miniconda.sh && \
+    /opt/conda/bin/conda install -y "python=${PYTHON_VERSION}" ipython && \
+    /opt/conda/bin/conda clean -ya
+
+FROM conda as conda-installs
+ARG PYTHON_VERSION=3.9
+ARG CUDA_VERSION=11.3
+ARG INSTALL_CHANNEL=pytorch
+RUN /opt/conda/bin/conda install -c "${INSTALL_CHANNEL}" -y "python=${PYTHON_VERSION}" pytorch torchvision torchtext "cudatoolkit=${CUDA_VERSION}" && \
```

Comment:
Can we also install `cudatoolkit-dev=${CUDA_VERSION}` from `conda-forge`, and `mpi4py` from regular conda?

You need those things for Deepspeed, or other libraries that ship fused GPU operations.

Reply:
Honestly I don't know why `cudatoolkit` doesn't just ship with the dev tools integrated. I think when you install CUDA from Nvidia's packages they all come together, so the conda people went out of their way to un-bundle it.

Reply:
Done in https://github.com/allenai/docker-images/pull/28/commits/9eedd8eeb25c9c89ccecd4ceee4765459e09b44d

Reply:
I've moved the base over to our common docker images in https://github.com/allenai/docker-images/pull/28 so that other projects can benefit from this.